app/src/main/java/org/myhush/silentdragon/MainActivity.kt

```diff
@@ -247,15 +284,27 @@ class MainActivity : AppCompatActivity(),
             if (confirmed.isNotEmpty()) {
                 var oddeven = "odd"
                 for (tx in confirmed) {
-                    fragTx.add(
-                        txList.id,
-                        TransactionItemFragment.newInstance(
-                            Klaxon().toJsonString(tx),
-                            oddeven
-                        ),
-                        "tag1"
-                    )
-                    oddeven = if (oddeven == "odd") "even" else "odd"
+                    if(tx.memo?.length == 0){
+                        fragTx.add(
+                            txList.id,
+                            TransactionItemFragment.newInstance(
+                                Klaxon().toJsonString(tx),
+                                oddeven
+                            ),
+                            "tag1"
+                        )
+                        oddeven = if (oddeven == "odd") "even" else "odd"
+                    } else {
+                        // test if contact exists
+                        if(Addressbook.findContactByInAddress(tx.addr) == null)
+                            Addressbook.addContact(fixName, tx.addr, fixAdd)
+                        // add message
+                        if(tx.type == "send")
+                            Addressbook.findContactByInAddress(tx.addr)!!.messageList.add(Message(tx.addr, tx, MessageType.SEND))
+                        else
```

Comment:
we should have many tx types to support future code. This code will break if old SDA versions have this and then we want to add a new tx type that potentially needs code in SD+SDL as well.